axelor-stock/src/main/java/com/axelor/apps/stock/service/StockMoveLineServiceImpl.java

```diff
@@ -1195,14 +1195,14 @@ public void setAvailableStatus(StockMoveLine stockMoveLine) {
 
   public List<TrackingNumber> getAvailableTrackingNumbers(
       StockMoveLine stockMoveLine, StockMove stockMove) {
-    List<TrackingNumber> trackingNumbers = null;
     String domain =
         "self.product.id = "
             + stockMoveLine.getProduct().getId()
-            + " AND self.id in (select stockLocationLine.trackingNumber.id from StockLocationLine stockLocationLine join StockLocation sl on sl.id = stockLocationLine.detailsStockLocation.id WHERE sl.id = "
+            + " AND self.id in (select stockLocationLine.trackingNumber.id from StockLocationLine stockLocationLine"
+            + " join StockLocation sl on sl.id = stockLocationLine.detailsStockLocation.id WHERE sl.id = "
             + stockMove.getFromStockLocation().getId()
-            + " )";
-    trackingNumbers = trackingNumberRepo.all().filter(domain).fetch();
+            + " AND coalesce(stockLocationLine.currentQty, 0) != 0)";
+    List<TrackingNumber> trackingNumbers = trackingNumberRepo.all().filter(domain).fetch();
```

Comment:
Small nitpick but you don't have to create a trackingNumbers variable and you can instead return trackingNumberRepo.all().filter(domain).fetch();